desktop/cmp/grid/impl/filter/headerfilter/HeaderFilterModel.ts

```diff
@@ -29,43 +39,52 @@ export class HeaderFilterModel extends HoistModel {
     @managed valuesTabModel: ValuesTabModel;
     @managed customTabModel: CustomTabModel;
 
-    get filterModel() {
+    get filterModel(): GridFilterModel {
         return this.parent.filterModel;
     }
 
-    get field() {
+    get field(): string {
         return this.fieldSpec.field;
     }
 
-    get store() {
+    get store(): Store {
         return this.filterModel.gridModel.store;
     }
 
-    get fieldType() {
+    get fieldType(): FieldType {
         return this.store.getField(this.field).type;
     }
 
-    get currentGridFilter() {
+    get currentGridFilter(): Filter {
         return this.filterModel.filter;
     }
 
-    get columnFilters() {
+    get columnFilters(): FieldFilter[] {
         return this.filterModel.getColumnFilters(this.field);
     }
 
-    get columnCompoundFilter() {
+    get columnCompoundFilter(): CompoundFilter {
         return this.filterModel.getColumnCompoundFilter(this.field);
     }
 
-    get hasFilter() {
+    get hasFilter(): boolean {
         return !isEmpty(this.columnFilters);
     }
 
-    get hasPendingFilter() {
+    get hasPendingFilter(): boolean {
+        return !!this.pendingFilter;
+    }
+
+    @computed
+    get isDirty(): boolean {
+        return !isEqual(this.columnFilters, flattenFilter(parseFilter(this.pendingFilter)));
```

Comment:
I'm actually a bit surprised that `isEqual` works with `Filter` instances, but it seems to.  We do have `Filter.equals()` - should / could we be using that?